spinoso-time/src/time/tzrs/error.rs

```diff
@@ -0,0 +1,82 @@
+use core::fmt;
+use std::error;
+
+use tz::error::{DateTimeError, ProjectDateTimeError, TzError};
+
+use super::offset::TzStringError;
+
+/// A wrapper around some of the errors provided by `tz-rs`.
+#[derive(Debug)]
+pub enum TimeErr {
+    /// Created when trying to create a DateTime, however the projection to a unix timestamp wasn't
+    /// achieveable. Generally thrown when exceeding the range of integers (e.g. `> i64::Max`).
+    ///
+    /// Note: This is just a wrapper over [`tz::error::ProjectDateTimeError`].
+    ProjectionError(ProjectDateTimeError),
+
+    /// Created when one of the parameters of a Datetime falls outside the allowed ranges (e.g.
+    /// 13th month, 32 day, 24th hour, etc)
+    ///
+    /// Note: [`tz::error::DateTimeError`] is only thrown from `tz-rs` when a provided component value is out of range.
+    ///
+    /// Note2: This is different from how MRI ruby is implemented. e.g. Second 60 is valid in MRI, and
+    /// will just add an additional second instead of erroring.
+    ComponentOutOfRangeError(DateTimeError),
+
+    /// The provided time zone string cannot be used
+    TzStringError(TzStringError),
+}
+
+impl error::Error for TimeErr {}
```

Comment:
let's add an impl for [`Error::source`]:

```suggestion
impl error::Error for TimeErr {
    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
        match self {
            Self:: ProjectionError(ref err) => Some(err),
            Self:: ComponentOutOfRangeError(ref err) => Some(err),
            Self:: TzStringError(ref err) => Some(err),
        }
    }
}
```

[`Error::source`]: https://doc.rust-lang.org/stable/std/error/trait.Error.html#method.source